api4/channel_test.go

```diff
@@ -2015,3 +2016,75 @@ func TestRemoveChannelMember(t *testing.T) {
 	_, resp = th.SystemAdminClient.RemoveUserFromChannel(privateChannel.Id, user2.Id)
 	CheckNoError(t, resp)
 }
+
+func TestAutocompleteChannels(t *testing.T) {
+	th := Setup().InitBasic()
+	defer th.TearDown()
+
+	// A private channel to make sure private channels are not used
+	utils.DisableDebugLogForTest()
+	ptown, _ := th.Client.CreateChannel(&model.Channel{
+		DisplayName: "Town",
+		Name:        "town",
+		Type:        model.CHANNEL_PRIVATE,
+		TeamId:      th.BasicTeam.Id,
+	})
+	utils.EnableDebugLogForTest()
+	defer func() {
+		th.Client.DeleteChannel(ptown.Id)
+	}()
+
+	for _, tc := range []struct {
+		description      string
+		teamId           string
+		fragment         string
+		expectedIncludes []string
+		expectedExcludes []string
+	}{
+		{
+			"Basic town-square",
+			th.BasicTeam.Id,
+			"town",
+			[]string{"town-square"},
+			[]string{"off-topic", "town"},
+		},
+		{
+			"Basic off-topic",
+			th.BasicTeam.Id,
+			"off-to",
+			[]string{"off-topic"},
+			[]string{"town-square", "town"},
+		},
+		{
+			"Basic town square and off topic",
+			th.BasicTeam.Id,
+			"to",
+			[]string{"off-topic", "town-square"},
+			[]string{"town"},
+		},
+	} {
+		if channels, resp := th.Client.AutocompleteChannelsForTeam(tc.teamId, tc.fragment); resp.Error != nil {
+			t.Fatal("Test case " + tc.description + " failed. Err: " + resp.Error.Error())
+		} else {
+			for _, expectedInclude := range tc.expectedIncludes {
```

Comment:
nit: Worth just collecting and sorting the found channel names and leveraging `assert.Equal` to dump the differences?

Reply:
Ya that's probably better. If this wasn't test code I would bother changing it.